strawberry/extensions/pyinstrument.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Iterator
+
+from pyinstrument import Profiler
+
+from strawberry.extensions.base_extension import SchemaExtension
+
+
+class PyInstrument(SchemaExtension):
+    """
+    Extension to profile the execution time of resolvers using PyInstrument.
+    """
+
+    def __init__(
+        self,
+        report_path: Path = Path("pyinstrument.html"),
+    ) -> None:
+        self._report_path = report_path
+
+    def on_operation(self) -> Iterator[None]:
+        """
+        Called when an operation is started,
+        in this case we start the profiler and yield
+        then we stop the profiler when the operation is done
+        """
+        # Start the profiler
+        profiler = Profiler()
+        profiler.start()
+
+        # Perform the operation
+        yield
+
+        # Stop the profiler
+        profiler.stop()
+        with Path.open(self._report_path, "w", encoding="utf-8") as f:
+            f.write(profiler.output_html())
```

Comment:
I think we can do `Path(self._report_path, "w", encoding="utf-8").write_text(...)`

Reply:
Another TIL. Will modify

Reply:
Looking into this the other option is to perform the whole operation without the `with` context however I'm unsure if that's a better option as it just turns these two lines into one larger line without the context manager. Is there a specific reason we may want to do this in one line?

Reply:
mostly to remove the `with` statement, since it is done inside `write_text` 😊

Reply:
ah, you can also remove `w`: https://github.com/python/cpython/blob/3.11/Lib/pathlib.py#L1070-L1079

Reply:
https://github.com/strawberry-graphql/strawberry/pull/2727/commits/b43ab659b032cfae5f1a413602fb2799632c2c67